controllers/resource/exposedPortHandler_test.go

```diff
@@ -0,0 +1,410 @@
+package resource
+
+import (
+	"context"
+	"github.com/cloudogu/cesapp-lib/core"
+	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	"github.com/cloudogu/k8s-dogu-operator/internal/cloudogu/mocks"
+	extMocks "github.com/cloudogu/k8s-dogu-operator/internal/thirdParty/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+func TestNewDoguExposedPortHandler(t *testing.T) {
+	// given
+	mockClient := fake.NewClientBuilder().Build()
+
+	// when
+	sut := NewDoguExposedPortHandler(mockClient)
+
+	// then
+	require.NotNil(t, sut)
+}
+
+func Test_doguExposedPortHandler_CreateOrUpdateCesLoadbalancerService(t *testing.T) {
+	t.Run("should return empty service and nil if dogu has no exposed ports", func(t *testing.T) {
+		// given
+		sut := &doguExposedPortHandler{}
+
+		// when
+		service, err := sut.CreateOrUpdateCesLoadbalancerService(context.TODO(), &k8sv1.Dogu{}, &core.Dogu{})
+
+		// then
+		require.Nil(t, err)
+		assert.Equal(t, service, &v1.Service{})
+	})
+
+	t.Run("should return error on error getting service error", func(t *testing.T) {
+		// given
+		nginxIngressCR := readNginxIngressDoguResource(t)
+		nginxIngressDogu := readNginxIngressDogu(t)
+		mockClient := extMocks.NewK8sClient(t)
+		mockClient.EXPECT().Get(context.TODO(), types.NamespacedName{Name: "ces-loadbalancer", Namespace: "ecosystem"}, &v1.Service{}).Return(assert.AnError)
+		sut := &doguExposedPortHandler{client: mockClient}
+
+		// when
+		_, err := sut.CreateOrUpdateCesLoadbalancerService(context.TODO(), nginxIngressCR, nginxIngressDogu)
+
+		// then
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "failed to get service ces-loadbalancer")
+		assert.ErrorIs(t, err, assert.AnError)
+	})
+
+	t.Run("should return an error on tcp/udp exposure if no loadbalancer is available", func(t *testing.T) {
+		// given
+		nginxIngressCR := readNginxIngressDoguResource(t)
+		nginxIngressDogu := readNginxIngressDogu(t)
+		serviceExposer := mocks.NewTcpUpdServiceExposer(t)
+		serviceExposer.EXPECT().ExposeOrUpdateDoguServices(context.TODO(), nginxIngressCR.Namespace, nginxIngressDogu).Return(assert.AnError)
```

Comment:
If you want to check what's happening when the loadbalancer is not available, shouldn't the mockClient.Get() function (which is used by deph.getCesLoadBalancerService(ctx, doguResource) in sut.CreateOrUpdateCesLoadbalancerService) return an error instead of the serviceExposer.ExposeOrUpdateDoguServices()?